lib/utils.py

```diff
@@ -141,22 +148,23 @@ def is_hostname(string):
         """
 
         try:
-            return bool(re.match("^(([a-zA-Z]|[a-zA-Z][a-zA-Z0-9\-]*[a-zA-Z0-9])\.)*([A-Za-z]|[A-Za-z][A-Za-z0-9\-]*[A-Za-z0-9])$", string))
+            return bool(re.match("^(([a-zA-Z]|[a-zA-Z][a-zA-Z0-9-]*[a-zA-Z0-9])\\.)*([A-Za-z]|[A-Za-z][A-Za-z0-9\-]*[A-Za-z0-9])$", string))
         except TypeError:
             return False
 
     @staticmethod
     def get_local_ipv4_address():
         """
-        Get's local ipv4 address
-        TODO: What if more than one interface present ?
+        Get local ipv4 address of the interface with the default gateway.
```

Comment:
Ich bin mir nicht sicher, dass 127.0.0.1 überall passt, wo diese Funktion genutzt wird. Siehst Du Vorteile bzw. Risiken?

Reply:
Das ist eine Änderung, die in der lib.network schon drin war, im Original wohl von Foxi. 

127.0.0.1 wird (fast) immer fehlerfrei funktionieren; ob es immer das ist, was der Nutzer will, ist die andere Frage.
Bei mehreren IP-Adressen ist es faktisch nicht möglich, die "richtige" automatisch zu ermitteln. Es gibt (ganz wenige) Fälle, wo das Netzwerk da ist, aber 127.0.0.1 nicht vergeben oder aktiv ist. 

Als default fallback ist das in den meisten Fällen jedenfalls besser als nichts. Ideal wäre anders, bzw.  ggf. nichts zurückzugeben. Damit werden die meisten aufrufenden Stellen aber nicht rechnen.

Reply:
Vorteil ist, wenn du z.B. eine VM oder einen physischen Computer ohne Netzwerk nutzt. Zum Entwickeln oder einfach nur zum testen. Oder SHNG auf einem Kleincomputer ohne Netzwerk IF nutzen willst um z.B. nur KNX (per USB) oder 1-Wire (per USB / serial) zu steuern. Das wäre denke ich so möglich ohne sonst irgendwas zu ändern da SHNG auf 127.0.0.1 zurückfällt was eigentlich ohne Probleme funktionieren müsste ? Oder sehe ich das falsch ?

Mit 127.0.0.1 läuft im Prinzip

Reply:
Im Wesentlichen ist das ok. Wenn kein Netzwerk konfiguriert ist, kann (nur *ix) 127.0.0.1 auch mal nicht verfügbar sein, aber dann schlägt das spätere Nutzen der IP sowieso fehl, egal, was man vorher macht.

Für mich ist 127.0.0.1 als Fallback ok.